respy/pre_processing/transition_matrix_exog_process.py

```diff
@@ -0,0 +1,170 @@
+        the exogenous process.
+    covariates : dictionary
+        A template for the specification of the covariates required for the exogenous
+        process.
+
+    """
+    # Check transition matrix conditions
+    _check_numerics(matrix)
+
+    # Make sure labels are strings
+    matrix.index = matrix.index.map(str)
+    matrix.columns = matrix.columns.map(str)
+
+    states = matrix.index
+    process_states = matrix.columns
+    # Create the covariates template
+    covariates = _create_covariates(states, process_name, process_states)
+    # Create logit values
+    transformed_matrix = _transform_matrix(matrix)
+    # Finally transfer to the params layout
+    params = _create_params(transformed_matrix, states, process_name, process_states)
+    return params, covariates
+
+
+def _create_params(transformed_matrix, states, process_name, process_states):
+    """Create the params entries for the respy initial files.
+
+    Create entries with logit coefficents instead of probabilities.
+    Parameters
+    ----------
+    transformed_matrix : pandas.DataFrame
+        The original DataFrame converted to logit values.
+    states : list
+        A list of the states on which probabilities for the exogenous process are
+        definded.
+    process_name : string
+        The name of the exogenous process.
+    process_states : list
+        A list of the possible outcomes of the exogenous process.
+
+    Returns
+    -------
+    params : pandas.DataFrame
+        A subset of the params initial file for respy. Contains all information on
+        the exogenous process.
+    """
+    # Create categories for the different states of the process
+    categories = [
+        f"exogenous_process_{process_name}_{state}" for state in process_states
+    ]
+    index = pd.MultiIndex.from_tuples(
+        itertools.product(categories, states), names=["category", "name"]
+    )
+    params = pd.DataFrame(index=index, columns=["value"], dtype="float")
+    # Transfer values
+    for process_state in process_states:
+        params.loc[
+            (f"exogenous_process_{process_name}_{process_state}", states), "value"
+        ] = transformed_matrix[process_state].values
+    return params
+
+
+def _transform_matrix(matrix):
+    """Transform probabilities to logit coefficients.
+
+    Parameters
+    ----------
+    matrix : pandas.DataFrame
+        The transition matrix for an exogenous process.
+    Returns
+    -------
+    transformed_matrix : pandas.DataFrame
+        The original DataFrame converted to logit values.
+
+    """
+    transformed_matrix = matrix.copy()
+    # Asign large negative numbers for zero probabilities
+    transformed_matrix[transformed_matrix == 0] = -1e300
+    transformed_matrix[transformed_matrix > 0] = np.log(
+        transformed_matrix[transformed_matrix > 0]
+    )
+    return transformed_matrix
+
+
+def _create_covariates(states, process_name, process_states):
```

Comment:
I think it would contribute to the clarity of the code base if we would rename this to _create_covariates_options or something since we already have a create_covariates function which assigns each state the vector of covariates!

Reply:
Renamed!